### PR TITLE
Feature/demand comment

### PR DESCRIPTION
**Pull request by mattdean-digicatapult**
Demand comment implementation. Note test coverage for demandA is slightly worse than demandB just because we'll deduplicating those soon hopefully.

### CODE REVIEW

File: src/controllers/v1/demandA/index.ts [hunk 1]
```diff
@@ -157,22 +165,109 @@ export class demandA extends Controller {
...
+
+  /**
+   * @summary Get a demandA creation transaction by ID
+   * @param demandAId The demandA's identifier
+   * @param creationId The demandA's creation ID
+   */
+  @Response<NotFound>(404, 'Item not found.')
+  @SuccessResponse('200')
+  @Get('{demandAId}/comment/{commentId}')
+  public async getDemandBComment(@Path() demandAId: UUID, commentId: UUID): Promise<TransactionResponse> {
```
Comment (jonmattgray) [inline src/controllers/v1/demandA/index.ts:237]: `getDemandAComment`

File: src/controllers/v1/demandA/index.ts [hunk 2]
```diff
@@ -187,3 +282,26 @@ const responseWithAlias = async (demandA: DemandRow): Promise<DemandResponse> =>
     updatedAt: demandA.updatedAt.toISOString(),
   }
 }
+
+const responseWithComments = async (
+  demandB: DemandResponse,
```
Comment (jonmattgray) [inline src/controllers/v1/demandA/index.ts:287]: `demandA`

File: src/lib/db/index.ts [hunk 3]
```diff
@@ -57,6 +67,16 @@ export interface DemandWithAttachmentRow {
   originalTokenId: number
 }
 
+const demandCommentsColumns = ['id', 'owner', 'state', 'attachment AS attachmentId', 'created_at AS createdAt']
+
+export interface DemandCommentRow {
```
Comment (jonmattgray) [inline src/lib/db/index.ts:72]: missing `updatedAt`?

File: src/lib/db/index.ts [hunk 4]
```diff
@@ -200,6 +244,14 @@ export default class Database {
       .where({ 'demand.id': id, subtype })
   }
 
+  insertDemandComment = async (comment: object) => {
+    return this.db().demand_comment().insert(comment).returning('*')
+  }
+
+  updateDemandComment = async (id: UUID, comment: object) => {
+    return this.db().demand_comment().update(comment).where({ id }).returning('*')
```
Comment (jonmattgray) [inline src/lib/db/index.ts:252]: missing `updated_at: this.client.fn.now()`

File: src/lib/db/index.ts (hunk 3 above)
Comment (mattdean-digicatapult) [reply src/lib/db/index.ts:72]: it is but we never use it. My preference is to leave it off now until we need it

File: src/lib/db/index.ts (hunk 4 above)
Comment (mattdean-digicatapult) [reply src/lib/db/index.ts]: yes

**Review by jonmattgray — APPROVED** [review]

**Pull request merged**